File: app/src/main/java/io/github/fanky10/android/mvpunittest/data/repository/impl/UsersRepositoryImpl.java
package io.github.fanky10.android.mvpunittest.data.repository.impl;

import java.util.ArrayList;
import java.util.List;

import io.github.fanky10.android.mvpunittest.data.entities.User;
import io.github.fanky10.android.mvpunittest.data.repository.UsersRepository;

public class UsersRepositoryImpl implements UsersRepository {
    @Override
    public List<User> getUsers() {
        List<User> users = new ArrayList<>();

        User mockUser = new User();
        mockUser.setUsername("admin@example.com");
        mockUser.setPassword("admin");
        users.add(mockUser);

        return users;
    }
}

File: app/src/main/java/io/github/fanky10/android/mvpunittest/data/repository/impl/UsersRepositoryImpl.kt
package io.github.fanky10.android.mvpunittest.data.repository.impl

import io.github.fanky10.android.mvpunittest.data.entities.User
import io.github.fanky10.android.mvpunittest.data.repository.UsersRepository
import java.util.*

class UsersRepositoryImpl : UsersRepository {
    override val users: List<User>
        get() {
            val users = ArrayList<User>()

            val mockUser = User()
            mockUser.username = "admin@example.com"
            mockUser.password = "admin"
            users.add(mockUser)

            return users
        }
}
